gc/base/HeapMap.hpp

```diff
@@ -255,6 +255,18 @@ class MM_HeapMap : public MM_BaseVirtual
 		return false;
 	}
 	
+	MMINLINE uintptr_t*
+	getSlotPtr(uintptr_t index) const
+	{
+		return &(getHeapMapBits()[index]);
+	}
```

Comment:
We can provide two versions of this function, the const-qualified version will ensure the heap map bits can't be mutated:

```suggestion
	MMINLINE uintptr_t *getSlotPtr(uintptr_t index) { return &(getHeapMapBits()[index]); }
	MMINLINE const uintptr_t *getSlotPtr(uintptr_t index) const { return &(getHeapMapBits()[index]); }
```